flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1902 @@
+    private static final String KEY_SEQUENCE = "key_sequence";
+    private static final String TABLE_SCHEMA = "table_schema";
+    private static final String UPDATE_RULE = "update_rule";
+    private static final String DELETE_RULE = "delete_rule";
+
+    private static final String TABLE_TYPE_TABLE = "TABLE";
+    private static final Duration FIXED_TICKET_EXPIRE_DURATION = Duration.ofMinutes(1);
+    private static final long QUERY_WATCHDOG_TIMEOUT_NANOS = Duration.ofSeconds(5).toNanos();
+
+    private static final Logger log = LoggerFactory.getLogger(FlightSqlResolver.class);
+
+    private static final KeyedObjectKey<ByteString, QueryBase> QUERY_KEY =
+            new KeyedObjectKey.BasicAdapter<>(QueryBase::handleId);
+
+    private static final KeyedObjectKey<ByteString, PreparedStatement> PREPARED_STATEMENT_KEY =
+            new KeyedObjectKey.BasicAdapter<>(PreparedStatement::handleId);
+
+    @VisibleForTesting
+    static final Schema DATASET_SCHEMA_SENTINEL = new Schema(List.of(Field.nullable("DO_NOT_USE", Utf8.INSTANCE)));
+
+    // Need dense_union support to implement this.
+    private static final UnsupportedCommand GET_SQL_INFO_HANDLER =
+            new UnsupportedCommand(CommandGetSqlInfo.getDescriptor(), CommandGetSqlInfo.class);
+    private static final UnsupportedCommand STATEMENT_UPDATE_HANDLER =
+            new UnsupportedCommand(CommandStatementUpdate.getDescriptor(), CommandStatementUpdate.class);
+    private static final UnsupportedCommand GET_CROSS_REFERENCE_HANDLER =
+            new UnsupportedCommand(CommandGetCrossReference.getDescriptor(), CommandGetCrossReference.class);
+    private static final UnsupportedCommand STATEMENT_SUBSTRAIT_PLAN_HANDLER =
+            new UnsupportedCommand(CommandStatementSubstraitPlan.getDescriptor(), CommandStatementSubstraitPlan.class);
+    private static final UnsupportedCommand PREPARED_STATEMENT_UPDATE_HANDLER = new UnsupportedCommand(
+            CommandPreparedStatementUpdate.getDescriptor(), CommandPreparedStatementUpdate.class);
+    private static final UnsupportedCommand GET_XDBC_TYPE_INFO_HANDLER =
+            new UnsupportedCommand(CommandGetXdbcTypeInfo.getDescriptor(), CommandGetXdbcTypeInfo.class);
+
+    // Unable to depends on TicketRouter, would be a circular dependency atm (since TicketRouter depends on all the
+    // TicketResolvers).
+    // private final TicketRouter router;
+    private final ScopeTicketResolver scopeTicketResolver;
+    private final ScheduledExecutorService scheduler;
+    private final Authorization authorization;
+    private final KeyedObjectHashMap<ByteString, QueryBase> queries;
+    private final KeyedObjectHashMap<ByteString, PreparedStatement> preparedStatements;
+
+    @Inject
+    public FlightSqlResolver(
+            final AuthorizationProvider authProvider,
+            final ScopeTicketResolver scopeTicketResolver,
+            final ScheduledExecutorService scheduler) {
```

Comment:
I believe my comments related to the scheduler, watchdog, and liveness may have gotten lost somewhere (or maybe I never made them)? I was thinking it might be nice to share some expiration related logic with session service and reuse the scheduler instead of a new ScheduledExecutorPool. It can be easier to target that approach with unit-tests related to expiry.

We all want another executor to go with the scheduler that knows anything it runs can't possibly block. (This would be nice, for example, for the web ui to establish new sessions even if the main executors are locked up performing large table operations.)

Reply:
One issue with `io.deephaven.server.util.Scheduler` is there is no way to cancel a command; not that it is something that is _necessary_, but when you provide a way to cancel you at least provide a proactive way for cleanup. Arguably, it would probably be easy to extend the Deephaven Scheduler to support this (since it mostly just wraps a ScheduledExecutorService).

Right now, FlightSQL prepared statements hook into the session lifecycle:

```java
this.session.addOnCloseCallback(onSessionClosedCallback = this::onSessionClosed);
```

On the other hand, the FlightSQL queries are much stricter and have a very short lifecycle - at least as I've setup here, a client has 5 seconds to initiate a resolve after the table has been created as part of the getFlightInfo (I will actually make this configurable now that I think about it), and so I've setup a watchdog w/ ScheduledExecutorService to auto-cancel after that timeout (and, you can argue that cleanup isn't as important since these are very short lived things).

I will reformulate w/ Deephaven scheduler to see how it looks.